golang/cosmos/x/swingset/keeper/keeper.go

```diff
@@ -144,8 +144,20 @@ func (k Keeper) SetStorage(ctx sdk.Context, path string, storage *types.Storage)
 	dataStore := prefix.NewStore(store, types.DataPrefix)
 	keysStore := prefix.NewStore(store, types.KeysPrefix)
 
+	// Update the value.
+	if storage.Value == "" {
+		dataStore.Delete([]byte(path))
+	} else {
+		dataStore.Set([]byte(path), k.cdc.MustMarshalLengthPrefixed(storage))
+	}
+
 	fullPathArray := strings.Split(path, ".")
 	oneUp := strings.Join(fullPathArray[0:len(fullPathArray)-1], ".")
+
+	if len(oneUp) == 0 {
```

Comment:
Consider `if oneUp == "" {`.

Reply:
I actually rewrote the SetStorage algorithm.  It should be good now (and I found a nondeterminism, too!).